docs/source/enterprise-guide/graph-security.mdx

```diff
@@ -0,0 +1,262 @@
+
+```javascript
+const server = ApolloServer({
+  gateway,
+  subscriptions: false,
+  context: ({ req }) => {
+    const userId = req.user?.sub || null;
+    return { userId };
+  }
+});
+```
+
+The token can then be forwarded as a header in requests to the subgraph services during query execution and the subgraphs can then authorize field usage as needed using a `RemoteGraphQLDataSource`:
+
+```javascript
+const gateway = new ApolloGateway({
+  // ...
+  buildService({ name, url }) {
+    return new RemoteGraphQLDataSource({
+      url,
+      willSendRequest({ request, context }) {
+        request.http.headers.set("user-id", context.userId);
+      }
+    });
+  }
+});
+```
+
+When taking this approach, a very important consideration here is network security. If you’re decoding and validating the token at the gateway and passing some of that data along as a header, then you will want to make sure that subgraphs aren’t directly accessible to the outside world. Ideally, communication between the gateway and subgraph services should be encrypted as well.
+
+Alternatively, you may opt for passing a token through and validating at the level of each subgraph. There will be a trade-off for security over performance with this approach because you may end up validating the same token multiple times in a single subgraph for particularly complex query plans, but that tradeoff may be worthwhile.
+
+## Security with managed federation
+
+Apart from protecting your GraphQL API from bad actors and locking down private data, you also need a window into how your API is being used and by whom to harden your GraphQL security posture. This is where a schema registry and observability tooling (such as [Apollo Studio](https://www.apollographql.com/docs/studio/)) come into play to help control who makes changes to your API and also monitor API usage and send alerts when something isn’t right.
+
+### Know who’s using your graph (and how)
+
+To enhance the utility of traces collected in your observability tooling, it's a best practice to require clients to identify themselves and that they consistently name operations. The web and mobile versions of Apollo Client make it easy to set custom headers for the name and version of a specific client to [segment traces in Apollo Studio](https://www.apollographql.com/docs/studio/client-awareness/and metrics by client). Other API clients can set the `apollographql-client-name` and `apollographql-client-version` request headers manually to provide client awareness as well. (As a bonus, client awareness also makes it easier to identify what clients may be impacted by a proposed breaking change to your API when running schema checks.)
```

Comment:
There's a syntax error that I've fixed, but I'm not sure if the sentence is what you expected it to be.

```suggestion
To enhance the utility of traces collected in your observability tooling, it's a best practice to require clients to identify themselves and that they consistently name operations. The web and mobile versions of Apollo Client make it easy to set custom headers for the name and version of a specific client to [segment traces and metrics in Apollo Studio by client](https://www.apollographql.com/docs/studio/client-awareness/). Other API clients can set the `apollographql-client-name` and `apollographql-client-version` request headers manually to provide client awareness as well. (As a bonus, client awareness also makes it easier to identify what clients may be impacted by a proposed breaking change to your API when running schema checks.)
```

Reply:
Good catch, fixed now.